lib/_http_outgoing.js

```diff
@@ -90,7 +90,7 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
+  if (typeof callback === 'function') {
```

Comment:
I had originally done this with a regex... and I am realizing now why you might want to check for callback... what if it doesn't exist :s